docs/library/uos.rst

```diff
@@ -43,18 +43,22 @@ Filesystem access
 
 .. function:: ilistdir([dir])
 
-   This function returns an iterator which then yields 3-tuples corresponding to
+   This function returns an iterator which then yields tuples corresponding to
    the entries in the directory that it is listing.  With no argument it lists the
    current directory, otherwise it lists the directory given by *dir*.
 
-   The 3-tuples have the form *(name, type, inode)*:
+   The tuples have the form *(name, type, inode[, size])*:
 
     - *name* is a string (or bytes if *dir* is a bytes object) and is the name of
       the entry;
     - *type* is an integer that specifies the type of the entry, with 0x4000 for
       directories and 0x8000 for regular files;
     - *inode* is an integer corresponding to the inode of the file, and may be 0
       for filesystems that don't have such a notion.
+    - Some platforms may return a 4-tuple that includes the entry's *size*.  For
+      file entries, *size* is an integer representing the size of the file
+      or -1 if unknown.  Its meaning is currently undefined for directory
+      entries, which should return -1.
```

Comment:
It's probably simpler to say that "Its meaning is currently undefined for directory entries.", that's it.  Then a port doesn't need to go to lengths to explicitly put in -1 for directories.  And it leaves it open to add a meaning later on.

Reply:
Updated and force pushed.